### ISSUE TITLE
tf.signal.kaiser_window loses the second derivative at beta zero

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#127245]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0, also reproduced on tf 2.22.0-dev20260904

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.13.5

### Current behavior?

`tf.signal.kaiser_window` returns an incorrect second derivative when `beta` is zero.

In the reproducer below, `beta = 3 + 2 * t`, and the input is evaluated at `t = -1.5`, so `beta` is exactly zero. The Kaiser window is defined using the modified Bessel function `I0`; since `I0` is an even entire function, the weighted Kaiser-window expression is smooth at `beta = 0`.

The forward value is computed correctly as `385.0`, but nested reverse-mode autodiff returns `0.0` for the second derivative with respect to `t`. The expected second derivative is approximately `-365.85185185185185`.

Expected behavior?

The second derivative of the weighted `tf.signal.kaiser_window` output at `t = -1.5` should be close to `-365.85185185185185`, not `0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""
os.environ["OMP_NUM_THREADS"] = "1"
os.environ["TF_NUM_INTRAOP_THREADS"] = "1"
os.environ["TF_NUM_INTEROP_THREADS"] = "1"

import tensorflow as tf

def target(t):
    y = tf.signal.kaiser_window(
        10,
        beta=3 + 2 * t,
        dtype=t.dtype,
    )
    weights = tf.constant(
        [1, 4, 9, 16, 25, 36, 49, 64, 81, 100],
        dtype=t.dtype,
    )
    return tf.reduce_sum(y * weights)

def reverse(fn):
    def grad(x):
        with tf.GradientTape() as tape:
            tape.watch(x)
            y = fn(x)
        return tape.gradient(y, x)
    return grad

x = tf.constant(-1.5, dtype=tf.float64)

forward = target(x)
actual = reverse(reverse(target))(x)
expected = -365.85185185185185

print("forward:", forward.numpy())
print("actual:", actual.numpy())
print("expected:", expected)

if abs(float(actual) - expected) > 1e-9:
    print("BUG REPRODUCED: tf.signal.kaiser_window loses the second derivative at beta zero")
else:
    print("not reproduced")
```

### Relevant log output

```shell
forward: 385.0
actual: 0.0
expected: -365.85185185185185
BUG REPRODUCED: tf.signal.kaiser_window loses the second derivative at beta zero
```

**Comment by Kayyuri** [comment]
Hi @ALinrunrun, Thank you for reporting this issue!

You've identified a bug where `tf.signal.kaiser_window` incorrectly computes the second derivative when the `beta` parameter is exactly zero. The `tf.math.bessel_i0e` function, which `kaiser_window` relies on, appears to have an issue with its higher-order gradients at `x=0`, causing the second derivative to incorrectly evaluate to `0.0`.

We’ll investigate the issue on our end and provide an update. Thanks!

**Comment by Venkat6871** [comment]
Hi @ALinrunrun ,
Thank you for raising your concern here.
A fix for this issue has been proposed in this [PR](https://github.com/tensorflow/tensorflow/pull/127549). Once it is merged, this issue will be resolved.

Hi @Phoebus-Liu ,
Thank you for raising the PR to address this issue. We truly appreciate your initiative and the effort in contributing the fix.
